corehq/apps/data_interfaces/models.py

```diff
@@ -340,18 +347,26 @@ def criteria_match(self, case, now):
                 (case.server_modified_on > (now - timedelta(days=self.server_modified_boundary))):
             return False
 
-        for criteria in self.memoized_criteria:
-            try:
-                result = criteria.definition.matches(case, now)
-            except CaseNotFound:
-                # This might happen if the criteria references a parent case and the
-                # parent case is not found
-                result = False
-
-            if not result:
-                return False
-
-        return True
+        if self.criteria_operator == 'ANY':
+            for criteria in self.memoized_criteria:
+                try:
+                    if criteria.definition.matches(case, now):
+                        return True
```

Comment:
I wonder if users are going to expect criteria to be processed in the order they appear on the page. It looks like the criteria don't have a guaranteed order. What would you think about enforcing ordering?

@snopoke @esoergel curious if you think this would make sense to do. I don't have a use case for a rule where the ordering would matter, I just think it's a question we'll get.

Reply:
Since the logic is commutative the ordering doesn't matter unless the criteria have side effects which they should not. If we introduced more complex logic operations then it could matter by YAGNI.